app/src/main/java/com/google/samples/apps/sunflower/PlantListFragment.kt

```diff
@@ -46,7 +47,11 @@ class PlantListFragment : Fragment() {
         val factory = InjectorUtils.providePlantListViewModelFactory(context)
         viewModel = ViewModelProviders.of(this, factory).get(PlantListViewModel::class.java)
 
-        val adapter = PlantAdapter()
+        val plantClickListener: (String) -> Unit = {
```

Comment:
nice 2 see, however, it doesn't change anything and bring benefits to the app, would you mind give some reasons why it should be extracted?

If I do, I'll create a `PlantItemViewModel` which impl. `OnClickListener` and holds `Plant`, meanwhile let each item of `RecyclerView` bind with the view-model, of cos, listener to `onClick` as well.

Reply:
You are right, the changes of this branch only move the click listener logic from adapter to fragment and don't bring benefits to the app. I have the changes you are suggesting in another branch. I will open another Pull Request and we could discuss there about the new approach.